File: Python/Assign Cookies.py
class Solution(object):
    def findContentChildren(self, g, s):
        """
        :type g: List[int]
        :type s: List[int]
        :rtype: int
        """
        g.sort()
        s.sort()
        i = 0
        counter = 0
        
        for c in g:
            for i in xrange(i, len(s)):
                if s[i] >= c:
                    counter += 1
                    i += 1
                    break
        return counter

File: cpp/Assign Cookies.cpp
class Solution {
public:
    int findContentChildren(vector<int>& g, vector<int>& s) {
        sort(g.begin(), g.end());
        sort(s.begin(),s.end());
        int counter = 0;
        auto g_pointer = g.begin();
        auto s_pointer = s.begin();
        while (g_pointer != g.end() && s_pointer != s.end()) {
            if (*s_pointer >= *g_pointer) {
                g_pointer++;
                counter++;
            }
            s_pointer++;
        }
        return counter;
    }
};
